crates/ruma-common/src/api/metadata.rs

```diff
@@ -96,6 +101,95 @@ impl Metadata {
 
         VersioningDecision::Unstable
     }
+
+    /// Generate the endpoint URL for this endpoint.
```

Comment:
For a public function, the documentation for this (and explaining the parameters) feel a little meager, I'd recommend hiding or adding some additional context and/or examples for how to use this.

Reply:
In the absence of a clear use case, I think this is going to be fine.